diagnostic.md

```diff
@@ -35,7 +35,9 @@ Record your responses inside the fenced code blocks below each question.
     ```
 
     ```md
-    <!-- your response here -->
+    I'm not completely sure on this one, but below is my best attempt.
+    1. The first route definition would be used for nested templates while the second one would be a better use case for what the above is actually accomplishing (doing a deeper view on a single item.  Basically a show instead of an index.).
+    2. The first one is setting up products/product and then mapping product to :product_id (effectively setting the path to products/:product_id).  The second route definition is doing that much more directly and just mapping product directly to /products/:product_id.
```

Comment:
You got it! 

The first `'product'` route definition is nested under the `'products'`
definition, whereas the second `'product'` route definition is a sibling of
the `'products'` definition.

Nested route definitions, like the first `'product'` route above, require
a nested file structure, such that Route and Template files for `'product'`
are expected to be found at `app/products/product`. For sibling routes,
folders are siblings as well.

The first `'product'` route definition instructs Ember to render
`app/products/product/template.hbs` in the context (the `outlet`) of the
parent `app/products/template.hbs`. The second `'product'` route definition
instructs Ember to render `app/product/template.hbs` in the context of the
application template.

These two route definitions do **not** have different paths.

1/1